engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -739,13 +745,21 @@ public static InetSocketAddress getRMWebAddress(Configuration conf, String rmId)
 
   public static InetSocketAddress getRMWebAddress(Configuration conf, boolean sslEnabled, String rmId)
   {
-    rmId = (rmId == null) ? "" : ("." + rmId);
+    boolean isHA = (rmId != null);
+    rmId = isHA ? ("." + rmId) : "";
+    if (isHA) {
+      conf = getYarnConfiguration(conf);
+      conf.set(ConfigUtils.RM_HA_ID, rmId);
+    }
     InetSocketAddress address;
     if (sslEnabled) {
       address = conf.getSocketAddr(YarnConfiguration.RM_WEBAPP_HTTPS_ADDRESS + rmId, YarnConfiguration.DEFAULT_RM_WEBAPP_HTTPS_ADDRESS, YarnConfiguration.DEFAULT_RM_WEBAPP_HTTPS_PORT);
```

Comment:
rmId should not be appended to `YarnConfiguration.RM_WEBAPP_HTTPS_ADDRESS`. `YarnConfiguration.getSocketAddr()` will return value based on `RM_HA_ID`.